### PR TITLE
Fixes parallel bug and updates to latest ci app specs.

### PR DESCRIPTION
**Pull request by tonyredondo**
This PR fixes a bug when running test in parallel caused by a read/write access to the internal CI App tags, also updates the CIApp specification.

### CODE REVIEW

File: internal/utils/ci_providers.go [hunk 1]
```diff
@@ -277,18 +285,33 @@ func extractGithubActions() map[string]string {
 		branch = branchOrTag
 	}
 
-	url := fmt.Sprintf("https://github.com/%s/commit/%s/checks", os.Getenv("GITHUB_REPOSITORY"), os.Getenv("GITHUB_SHA"))
+	serverUrl := os.Getenv("GITHUB_SERVER_URL")
+	if serverUrl == "" {
+		serverUrl = "https://github.com"
+	}
+
+	rawRepository := fmt.Sprintf("%s/%s", serverUrl, os.Getenv("GITHUB_REPOSITORY"))
```
Comment (AdrianLC) [inline internal/utils/ci_providers.go:293]: this is not save if they provide `GITHUB_SERVER_URL` with a leading slash

File: internal/utils/ci_providers.go [hunk 2]
```diff
@@ -277,18 +285,33 @@ func extractGithubActions() map[string]string {
...
+	pipelineId := os.Getenv("GITHUB_RUN_ID")
+	commitSha := os.Getenv("GITHUB_SHA")
+
 	tags[constants.CIProviderName] = "github"
-	tags[constants.GitRepositoryURL] = fmt.Sprintf("https://github.com/%s.git", os.Getenv("GITHUB_REPOSITORY"))
-	tags[constants.GitCommitSHA] = os.Getenv("GITHUB_SHA")
+	tags[constants.GitRepositoryURL] = fmt.Sprintf("%s.git", rawRepository)
```
Comment (AdrianLC) [inline internal/utils/ci_providers.go:298]:
```suggestion
	tags[constants.GitRepositoryURL] = rawRepository+".git"
```

File: internal/utils/ci_providers.go [hunk 3]
```diff
@@ -277,18 +285,33 @@ func extractGithubActions() map[string]string {
...
-	tags[constants.CIPipelineURL] = url
-	tags[constants.CIJobURL] = url
+	tags[constants.CIJobURL] = fmt.Sprintf("%s/commit/%s/checks", rawRepository, commitSha)
+
+	attempts := os.Getenv("GITHUB_RUN_ATTEMPT")
+	if attempts == "" {
+		tags[constants.CIPipelineURL] = fmt.Sprintf("%s/actions/runs/%s", rawRepository, pipelineId)
+	} else {
+		tags[constants.CIPipelineURL] = fmt.Sprintf("%s/actions/runs/%s/attempts/%s", rawRepository, pipelineId, attempts)
+	}
```
Comment (AdrianLC) [inline internal/utils/ci_providers.go:313]: same concern about the leading slash for these

File: option.go [hunk 4]
```diff
@@ -38,65 +40,92 @@ func defaults(cfg *config) {
...
 		}
-		if _, ok := tags[constants.GitCommitMessage]; !ok {
-			tags[constants.GitCommitMessage] = gitData.CommitMessage
+		if _, ok := localTags[constants.GitCommitMessage]; !ok {
+			localTags[constants.GitCommitMessage] = gitData.CommitMessage
 		}
 	}
+
+	// Replace global tags with local copy
+	tags = localTags
```
Comment (AdrianLC) [inline option.go:108]: can you lock only when doing this replacement?

File: internal/utils/ci_providers.go (hunk 1 above)
Comment (tonyredondo) [reply internal/utils/ci_providers.go:294]: fixed in 334a193

File: internal/utils/ci_providers.go (hunk 2 above)
Comment (tonyredondo) [reply internal/utils/ci_providers.go]: fixed in 334a193

File: internal/utils/ci_providers.go (hunk 3 above)
Comment (tonyredondo) [reply internal/utils/ci_providers.go:314]: fixed in 334a193

File: option.go (hunk 4 above)
Comment (tonyredondo) [reply option.go:108]: fixed in 334a193

**Review by AdrianLC — APPROVED** [review]
LGTM

**Pull request merged**